### ISSUE TITLE
Forward ref for checkboxes

### ISSUE THREAD

**Issue by brightpixels** [surevine/govuk-react-jsx#45]
Hello,
I am trying to use checkboxes (https://surevine.github.io/govuk-react-jsx/?path=/story/checkboxes--with-optional-form-group-classes-showing-group-error) and with React-hook-form (https://react-hook-form.com/get-started/).

I am having problems passing a ref attribute and getting this error: 
````
index.js:1 Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
````

I am trying to do this:
````

        <Checkboxes
          items={[
            {
              children: ['Email'],
              conditional: {
                children: [
                  <label key="0" className="govuk-label" htmlFor="context-email">
                    Email address
                  </label>,
                  <span key="1" className="govuk-error-message" id="contact-phone-error">
                    Problem with input
                  </span>,
                  <input ref={register({required: true})} key="2" className="govuk-input govuk-!-width-one-third" id="context-email" name="context-email" type="text" />,
                ],
              },
              value: 'email',
            },
          ]}
          name="email"
        />
````

Adding `ref={register({required: true})}` to other components like `Input` works before you are forwarding the refs but don't appear to be doing it for checkboxes (and probably radio buttons)

**Comment by andymantell** [comment]
@brightpixels I don't use forwardRef on the checkboxes, radios and date input components because they can represent multiple form elements. So instead of taking a top level ref prop, the individual items take a ref. So in your example, you could put a ref in the first object inside the items array :
```
      const ref = useRef();

      <Checkboxes
          items={[
            {
              ref: ref,
              children: ['Email'],
              conditional: {
                children: [
                  <label key="0" className="govuk-label" htmlFor="context-email">
                    Email address
                  </label>,
                  <span key="1" className="govuk-error-message" id="contact-phone-error">
                    Problem with input
                  </span>,
                  <input key="2" className="govuk-input govuk-!-width-one-third" id="context-email" name="context-email" type="text" />,
                ],
              },
              value: 'email',
            },
          ]}
          name="email"
        />
```
This would get you a reference to the checkbox.

Also worth noting, the demo code from the storybook shows rendered output being sent as conditional children, but you can also pass full JSX here - i.e. actual <Input /> components

**Comment by andymantell** [comment]
Also, please make sure you are using the latest version if you aren't already - forwardRef only got introduced in the latest 1.3.0

**Comment by brightpixels** [comment]
Thank you.

**Comment by brightpixels** [comment]
If there were several items (e.g. email, phone, etc), should  They take the same ref? Specially for a radio because from a set of radio options, there should be just a single value.

**Comment by brightpixels** [comment]
"Also worth noting, the demo code from the storybook shows rendered output being sent as conditional children, but you can also pass full JSX here - i.e. actual components"

Just to be clear, are you talking about the `children` property under `conditional` property? If so, that's handy to know.

**Comment by andymantell** [comment]
I'll try and put together a radios demo with react-hook-form and see what happens. Multiple elements cannot have the same ref.

As for the children property - yes, that is true of all `children` props - they can all take JSX. The reason it's rendered as html there is because the contents for those demos is coming from the govuk examples which is in html. I'm gonna log a new issue to consider doing this

**Comment by andymantell** [comment]
@brightpixels I've had a quick look at this and there's some issues with govuk-react-jsx and react-hook-form - I've started looking at what I can do to be compatible with it. This package has been built assuming that people would be using controlled inputs whereas react-hook-form uses uncontrolled inputs. It wasn't a conscious design decision - just that the project I was working on using this used Formik which uses controlled inputs.

I'll see what I can do, bear with me.

**Comment by andymantell** [comment]
I fixed a couple of issues around the radios in #48 and released govuk-react-jsx@1.4.0. Everything else seemed to be ok. I've not tackled the date input yet with react-hook-form as it's a bit more fiddly. I'll take that on in #4 

You can find the example code here:

https://github.com/surevine/govuk-react-jsx-examples/blob/master/src/forms/ReactHookForm.js

Which you can see running here:

https://surevine.github.io/govuk-react-jsx-examples/forms/react-hook-form

Any more problems let me know. Maybe reopen #48 or open a new ticket instead of carrying on this thread as we've strayed a bit from the original topic!
